lib/pbench/server/api/__init__.py

```diff
@@ -111,6 +113,9 @@ def register_endpoints(api, app, config):
     api.add_resource(
         Login, f"{base_uri}/login", resource_class_args=(config, logger, token_auth),
     )
+    api.add_resource(
+        ForgotPassword, f"{base_uri}/forgot_password", resource_class_args=(config, mail, logger, token_auth),
```

Comment:
Pass `app` instead of `mailer` here; that'll keep the `Mail` package encapsulated where it's used. And the only change you'd have to make to this file would be your `api.add_resource()` call since `app` is already available inside `register_endpoints`. If possible, I'd prefer not to have this registration mechanism get special cased for various APIs...